### DISCUSSION TITLE
Self-hosting Directus fork

### DISCUSSION THREAD

**Discussion by jroxendal** [directus/directus#16113] [Archive]
### Describe the Bug

I'm trying to self-host my fork of Directus using the docs: https://docs.directus.io/contributing/running-locally.html
Everything works fine if I run using `SERVE_APP=true`, but when I run `pnpm --filter app build && pnpm --filter app serve` and run the API with `SERVE_APP=false` I cannot get the parts to communicate properly. I gather from this issue:
https://github.com/directus/directus/issues/8107
that I'm supposed to access Directus through the app port of 4173, and it works for a while but crashes hard on browser refresh with tons of 404s like `http://localhost:4173/admin/content/test/assets/index.d5d6e2a0.entry.js net::ERR_ABORTED 404 (Not Found)`

If I use `SERVE_APP=true` I don't see my code modifications from my fork. Is there some way to specify which version of the app that is served through SERVE_APP?

### To Reproduce

1. Fork Directus
2. Commit some code change
3. Clone and build
4. `pnmp --filter app serve` and `pnpm --filter directus start`
5. Access http://localhost:4173/admin/login
6. Go to some content and hit refresh

### Errors Shown

`http://localhost:4173/admin/content/test/assets/vue-router.4ad81e1d.entry.js net::ERR_ABORTED 404 (Not Found)`

**Comment by br41nslug** [comment]
The running locally docs are meant for debugging / dev purposes and require you to run the 2 parts separately `pnpm -F directus dev` and `pnpm -F @directus/app dev`. This setup is however not terribly friendly to be hosted with Vite HMR and all that stuff if you'd like to run the app in production instead like `npx directus start` you can use `pnpm -F directus cli start`.

**Reply by jroxendal** [reply]
Thanks for the reply. I did finally get it working when changing the base variable in the vite config to `/admin` instead of just `/`. Now I can access the vite preview server at port 4173 and it communicates properly with the api started using `SERVE_APP=false pnpm -F directus start`. 

However, this method _fails_ to load my custom extension. When adding a custom extension using a normal setup, I simply do `cd api && pnpm install [github-extension-repo-url]` to make it available. Not sure how to install the extension using the method above. Do you know?

**Reply by br41nslug** [reply]
not quite sure tbh why it wouldnt be picking up on the npm extensions, i've exclusively used local extensions so far.
